lib/protocol/protocol.go

```diff
@@ -182,11 +182,12 @@ type rawConnection struct {
 	nextID    int32
 	nextIDMut sync.Mutex
 
-	outbox        chan asyncMessage
-	closed        chan struct{}
-	closeOnce     sync.Once
-	sendCloseOnce sync.Once
-	compression   Compression
+	outbox           chan asyncMessage
+	clusterConfigBox chan *ClusterConfig
```

Comment:
why pointer?

Reply:
It needs to implement the `message` interface which only the pointer does.